### PR TITLE
add protected for methods checkWaitReadTime and informReadOperation

### PR DESCRIPTION
**Pull request by brunoroberto**
I added protected for methods checkWaitReadTime and informReadOperation to allow override methods as an open issue #5155 .

### CODE REVIEW

File: handler/src/main/java/io/netty/handler/traffic/AbstractTrafficShapingHandler.java [hunk 1]
```diff
@@ -512,20 +512,22 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) throw
 
     /**
      * Method overridden in GTSH to take into account specific timer for the channel.
+     * @param ctx
      * @param wait the wait delay computed in ms
      * @param now the relative now time in ms
      * @return the wait to use according to the context
      */
-    long checkWaitReadTime(final ChannelHandlerContext ctx, long wait, final long now) {
+    protected long checkWaitReadTime(final ChannelHandlerContext ctx, long wait, final long now) {
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/traffic/AbstractTrafficShapingHandler.java]: MAJOR Remove this unused method parameter "ctx". [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1172)

**Comment by Scottmitch** [comment]
@fredericBregier - WDYT?

**Comment by Scottmitch** [comment]
@brunoroberto - Please sign the [ICLA](https://docs.google.com/spreadsheet/viewform?formkey=dHBjc1YzdWhsZERUQnhlSklsbG1KT1E6MQ) and follow the commit guidelines in http://netty.io/wiki/developer-guide.html

**Comment by normanmaurer** [comment]
@brunoroberto @fredericBregier ping

**Comment by fredericBregier** [comment]
Sorry to come back late...

I believe initially we design this to be package protected, not globally protected. They were originally made to allow `GlobalTrafficShapingHandler` to extend them, within the same package, meaning that it can allow it to access to protected attributed and methods, including package protected ones, which are pretty much used. This was meant to protect the implementation as it is a bit complex and could be misused.

If you think is it better to be able to extend those methods (which are made to be extended) including out of the package (meaning one is not obliged to extend within the same package), then it's fine for me to add protected to them.
You might consider however 2 points:
- Is that enough ? Menaing that's all you need ? What about other package protected methods/attributes ?
- What's the difficulty to create another TrafficShaping handler within the very same package for your own need ? Is that breaking some API rules ?

**Comment by normanmaurer** [comment]
Let me close this until @brunoroberto still has the demand... If so please reopen

**Pull request closed without merging**